gpiozero/pins/pi.py

```diff
@@ -217,6 +247,98 @@ def _get_spi_class(self, shared, hardware):
         SPI device(s) rather than a bit-banged software implementation.
         """
         raise NotImplementedError
+    
+    def i2c(self, **i2c_args):
+        """
+        Returns an I2C interface, for the specified I2C *port*, or for the
+        specified pins (*data_pin* and *clock_pin*).  Only one of the schemes
+        can be used; attempting to mix *port* with pin numbers will raise
+        :exc:`~gpiozero.I2CBadArgs`.
+        """
+        i2c_args, kwargs = self._extract_i2c_args(**i2c_args)
+        shared = bool(kwargs.pop('shared', False))
+        if kwargs:
+            raise I2CBadArgs(
+                'unrecognized keyword argument {arg}'.format(
+                    arg=kwargs.popitem()[0]))
+        try:
+            return self._get_i2c_class(shared, hardware=True)(
+                pin_factory=self, **i2c_args)
+        except Exception as e:
+            warnings.warn(
+                I2CSoftwareFallback(
+                    'failed to initialize hardware I2C, falling back to '
+                    'software (error was: {e!s})'.format(e=e)))
+        return self._get_i2c_class(shared, hardware=False)(
+            pin_factory=self, **i2c_args)
+    
+    def _extract_i2c_args(self, **kwargs):
+        """
+        Given a set of keyword arguments, splits it into those relevant to I2C
+        implementations and all the rest. I2C arguments are augmented with
+        defaults and converted into the pin format (from the port format) if
+        necessary.
+
+        Returns a tuple of ``(i2c_args, other_args)``.
+        """
+        try:
+            device = kwargs.pop("device")
+        except KeyError:
+            raise I2CBadArgs(
+                'device address is not specified as the *device* keyword'
```

Comment:
Looks like this `device` kwarg isn't documented? Is it the hardware-ID of the particular device on the I2C bus?

How does your I2C implementation cope with multiple different devices (with different hardware addresses) being connected to the same I2C bus?

Reply:
The `device` kwarg is the device address on the I2C bus. Completely forgot to add this in the docstring.